python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py

```diff
@@ -0,0 +1,464 @@
+#!/usr/bin/python
+import logging
+from datetime import timedelta
+
+import numpy as np
+import pandas as pd
+from astropy import units as u
+from astropy.time import Time
+from lsst.summit.utils.efdUtils import getEfdData
+from lsst.summit.utils.tmaUtils import TMAEvent, TMAEventMaker
+from plots import inertia_compensation_system
+
+# Configure the logger
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+formatter.datefmt = "%Y-%m-%d %H:%M:%S"
+
+handler = logging.StreamHandler()
+handler.setLevel(logging.ERROR)
+handler.setFormatter(formatter)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(handler)
+
+# TMAEventMaker needs to be instantiated only once.
+event_maker = TMAEventMaker()
+
+__all__ = ["ICSAnalysis"]
+
+
+class ICSAnalysis:
+    """
+    Evaluate the M1M3 Inertia Compensation System's performance by
+    calculating the minima, maxima and peak-to-peak values during a
+    slew. In addition, calculates the mean, median and standard deviation
+    when the slew has contant velocity or zero acceleration.
+
+    Parameters
+    ----------
+    event : `lsst.summit.utils.tmaUtils.TMAEvent`
+        Abtract representation of a slew event.
+    """
+
+    def __init__(
+        self,
+        event: TMAEvent,
+        inner_pad: float = 0.0,
+        outer_pad: float = 0.0,
+        n_sigma: float = 1.0,
+    ):
+        self.event = event
+        self.inner_pad = inner_pad * u.second
+        self.outer_pad = outer_pad * u.second
+        self.n_sigma = n_sigma
+        self.client = event_maker.client
+
+        # TODO: Find a better way to implement the logger inside the class
+        self.logger = logger
```

Comment:
Loggers are cheap - there's no harm in putting the logger creation into the class itself, and indeed, it should not be being done on import like it is now, so that's a win in both places. Feel free to make a `self._makeLogger()` function to stick it all in there though, and just call that from the `__init__`, that's fine.